src/main/java/com/cburch/logisim/util/LineBuffer.java

```diff
@@ -547,62 +572,75 @@ public LineBuffer addRemarkBlock(String remarkText, Integer nrOfIndentSpaces) {
    */
   protected ArrayList<String> buildRemarkBlock(String remarkText, Integer nrOfIndentSpaces) {
     final var maxRemarkLength = MAX_LINE_LENGTH - 2 * Hdl.remarkOverhead() - nrOfIndentSpaces;
-    final var remarkWords = remarkText.split(" ");
+    final var remarkLines = remarkText.split("\n");
     final var oneLine = new StringBuilder();
     final var contents = new ArrayList<String>();
-    var maxWordLength = 0;
-    for (final var word : remarkWords) {
-      if (word.length() > maxWordLength) maxWordLength = word.length();
-    }
-    if (maxRemarkLength < maxWordLength) return contents;
     /* we start with generating the first remark line */
     while (oneLine.length() < nrOfIndentSpaces) oneLine.append(" ");
     for (var i = 0; i < MAX_LINE_LENGTH - nrOfIndentSpaces; i++) {
-      oneLine.append(Hdl.getRemakrChar(i == 0, i == MAX_LINE_LENGTH - nrOfIndentSpaces - 1));
+      oneLine.append(Hdl.getRemarkChar(i == 0, i == MAX_LINE_LENGTH - nrOfIndentSpaces - 1));
     }
     contents.add(oneLine.toString());
     oneLine.setLength(0);
-    /* Next we put the remark text block in 1 or multiple lines */
-    for (final var remarkWord : remarkWords) {
-      if ((oneLine.length() + remarkWord.length() + Hdl.remarkOverhead()) > (MAX_LINE_LENGTH - 1)) {
-        /* Next word does not fit, we end this line and create a new one */
+    for (int lineIndex = 0; lineIndex < remarkLines.length; lineIndex++) {
+      final var remarkWords = remarkLines[lineIndex].split(" ");
+      var maxWordLength = 0;
+      for (final var word : remarkWords) {
+        if (word.length() > maxWordLength) maxWordLength = word.length();
```

Comment:
This line can be simplified:

```java
maxWordLength = Math.max(maxWordLength, word.length());
```

which is clearer in intent and saves us one call.

Reply:
done